pyuvdata/uvdata/ms.py

```diff
@@ -1165,6 +1165,17 @@ def write_ms(
 
             ms.putcol("FIELD_ID", np.repeat(field_id_array, self.Nspws))
 
+        # If scan numbers are not already defined from reading an MS,
+        # group integrations (rows) into scan numbers.
+        if self.scan_number_array is None:
+            self._set_scan_numbers()
+
+        if self.multi_phase_center:
```

Comment:
I'm a little confused as to what this block of code is doing -- why the dependence on `self.multi_phase_center`? If anything, I would have expected it to be if `self.flex_spw` or `self.Nspws != 1` (or even forgoing the if/then statement all together, since the code below should work if you only have 1 spectral window).

Reply:
Thanks for catching this! Agreed, this was wrong. I think it was leftover from when this code was in the above `if self.multi_phase_center` block above.

I've swapped to `self.Nspws > 1`. I could drop it if preferred, though I figured skipping a redundant `np.repeat` call is slightly more efficient. What do you think?

Reply:
I'm reasonably content either way, just as long as `self.multi_phase_center` isn't in there anymore 😉  Let's keep it as `self.Nspws > 1`.